catalog/imaging/transformer/catalog-transformer-nitf/src/test/java/ddf/catalog/transformer/nitf/TestNitfInputTransformer.java

```diff
@@ -0,0 +1,176 @@
+                .getContextClassLoader()
+                .getResourceAsStream("nitf.xml");
+        reader.parseMetacardDefinition(is);
+        metacardFactory = mf;
+        System.setProperty("java.awt.headless", "true");
+        Toolkit tk = Toolkit.getDefaultToolkit();
+    }
+
+    @Before
+    public void createTransformer()
+            throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+        transformer = new NitfInputTransformer();
+        CatalogFramework catalog = mock(CatalogFramework.class);
+        when(catalog.query(Matchers.any(QueryRequest.class))).thenReturn(new QueryResponseImpl(null,
+                "sourceId"));
+        transformer.setMetacardFactory(metacardFactory);
+    }
+
+    @Test(expected = CatalogTransformerException.class)
+    public void testNullInput()
+            throws IOException, CatalogTransformerException, UnsupportedQueryException,
+            SourceUnavailableException, FederationException {
+        transformer.transform(null);
+    }
+
+    @Test(expected = CatalogTransformerException.class)
+    public void testBadInput()
+            throws IOException, CatalogTransformerException, UnsupportedQueryException,
+            SourceUnavailableException, FederationException {
+        transformer.transform(new ByteArrayInputStream("{key=".getBytes()));
+    }
+
+    @Test
+    public void testSorcerWithBE()
+            throws IOException, CatalogTransformerException, UnsupportedQueryException,
+            SourceUnavailableException, FederationException, ParseException {
+        Metacard metacard = transformer.transform(getInputStream(BE_NUM_NITF));
+
+        assertNotNull(metacard);
+
+        validateDate(metacard, metacard.getCreatedDate(), "2014-08-17 07:22:41");
+    }
+
+    @Test
+    public void testTreParsing()
+            throws IOException, CatalogTransformerException, UnsupportedQueryException,
+            SourceUnavailableException, FederationException, ParseException {
+        Metacard metacard = transformer.transform(getInputStream(TRE_NITF));
+
+        assertNotNull(metacard);
+
+        validateDate(metacard, metacard.getCreatedDate(), "1999-02-10 14:01:44");
+    }
+
+    @Test
+    public void testNitfParsing()
+            throws IOException, CatalogTransformerException, UnsupportedQueryException,
+            SourceUnavailableException, FederationException, ParseException {
+        Metacard metacard = transformer.transform(getInputStream(GEO_NITF));
+
+        assertNotNull(metacard);
+
+        validateDate(metacard, metacard.getCreatedDate(), "1997-12-17 10:26:30");
+        validateDate(metacard, metacard.getEffectiveDate(), "1997-12-17 10:26:30");
+        validateDate(metacard, metacard.getModifiedDate(), "1997-12-17 10:26:30");
+        //validateDate(metacard, metacard.getExpirationDate(), "1997-12-17 10:26:30");
+        assertThat(Nitf.NAME, is(metacard.getMetacardType().getName()));
```

Comment:
I think it makes more sense to use the pattern `assertThat(valueToTest, is(expectedValue))`.